examples/meta/generator/targets/cpp.json

```diff
@@ -101,7 +101,7 @@
         "StaticCall": "C$typeName::$method($arguments)",
         "GlobalCall": "$method($arguments)",
         "Identifier": "$identifier",
-        "Enum":"(machine_int_t)$typeName::$value"
```

Comment:
and this does not break other examples? I cannot remember why exactly this case was here, but I remember putting in there for a reason...
`ctest -R generated_cpp` should bring clarity

Reply:
I believe you put that to make meta_api-enums work. Removing machine_int_t will cause the example to fail. However, to make the example I added to work you have to remove machine_int_t. I was hoping to see the CI results together and discuss how to proceed.

I prefer to not have to convert the enum types.

This is the output I get:

ctest -R generated_cpp-meta_api-enums -V
UpdateCTestConfiguration  from :/home/elavram/Workfolder/shogun/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/elavram/Workfolder/shogun/build/DartConfiguration.tcl
Test project /home/elavram/Workfolder/shogun/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 47
    Start 47: generated_cpp-meta_api-enums

47: Test command: /home/elavram/Workfolder/shogun/build/examples/meta/cpp/meta_api/cpp-meta_api-enums
47: Test timeout computed to be: 9.99988e+06
47: [ERROR] In file /home/elavram/Workfolder/shogun/src/shogun/base/SGObject.h line 366: Cannot put parameter LibLinear::liblinear_solver_type of type int, incompatible provided type shogun::LIBLINEAR_SOLVER_TYPE.
47: terminate called after throwing an instance of 'shogun::ShogunException'
47:   what():  [ERROR] In file /home/elavram/Workfolder/shogun/src/shogun/base/SGObject.h line 366: Cannot put parameter LibLinear::liblinear_solver_type of type int, incompatible provided type shogun::LIBLINEAR_SOLVER_TYPE.
47: 
1/1 Test #47: generated_cpp-meta_api-enums .....***Exception: Child aborted  0.16 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.18 sec

The following tests FAILED:
	 47 - generated_cpp-meta_api-enums (Child aborted)
Errors while running CTest

Reply:
I think a solution to consider is to convert the all "enum" to "enum class" so there will be no automatic convert to int and it will be safer.